h/services/annotation_json_presentation/service.py

```diff
@@ -94,7 +72,8 @@ def _get_user_dependent_content(self, annotation, user):
             }
 
         # The hidden value depends on whether you are the author
-        if not annotation.is_hidden or self._user_is_author(user, annotation):
+        user_is_author = user and user.userid == annotation.userid
+        if user_is_author or not annotation.is_hidden:
```

Comment:
Maybe this could be assigned to ` model["hidden"]` it might look hideous if `not` is required.

Reply:
Actually I think you're totally right here. An if statement which sets a boolean?

I suspect this is a case of not spotting the refactoring trees for the woods. This has been a few steps to get this far

I think this is actually:

```python
# Authors don't see their annotations as hidden even if they are
model["hidden"] = annotation.is_hidden and not user_is_author
```

Reply:
I've had a go but there's more than just the bool in the conditional. There's another conditional in there as well which would require repeating the condition.

So the "True" side of the conditional isn't just that it's hidden, but that it's hidden and the user isn't the author. I'm not sure there's a shorter way to do it?